src/parameters.rs

```diff
@@ -1,16 +1,73 @@
 use std::fmt::Debug;
+use std::path::PathBuf;
 
-use ixa::define_global_property;
+use ixa::{define_global_property, error::IxaError};
 use serde::{Deserialize, Serialize};
 
 #[derive(Serialize, Deserialize, Debug, Clone)]
 pub struct ParametersValues {
-    pub population: usize,
     pub max_time: f64,
     pub seed: u64,
     pub r_0: f64,
     pub infection_duration: f64,
     pub generation_interval: f64,
     pub report_period: f64,
+    pub synth_population_file: PathBuf,
+    pub population_periodic_report: String,
+}
+
+fn validate_inputs(parameters: &ParametersValues) -> Result<(), IxaError> {
+    if parameters.r_0 < 0.0 {
+        return Err(IxaError::IxaError(
+            "r_0 must be a non-negative number.".to_string(),
+        ));
+    }
+    if parameters.generation_interval <= 0.0 {
+        return Err(IxaError::IxaError(
+            "The generation interval must be positive.".to_string(),
+        ));
+    }
+    Ok(())
+}
+
+define_global_property!(Parameters, ParametersValues, |p| { validate_inputs(p) });
```

Comment:
```suggestion
define_global_property!(Parameters, ParametersValues, validate_inputs );
```

You don't need a closure here because `validate_inputs()` already has the right signature.